lib/object-mask.js

```diff
@@ -396,34 +405,30 @@ function invert(mask) {
 	return result;
 }
 
-// Throws on (deep) attempt to subtract collection from scalar
 function subtract(a, b) {
 	// base cases:
+	if (a === true) return invert(b);
+	if (a === false) return false;
+	if (!a) return invert(b);
 	if (!b) return a;
-	if (!a || b === true || _.isEqual(a, b)) return false;
-	if (objtools.isScalar(a) && !objtools.isScalar(b)) {
-		throw new XError(XError.INVALID_ARGUMENT, {
-			message: 'cannot subtract a non-scalar from a scalar',
-			data: { a: a, b: b }
-		});
-	}
+	if (b === true || _.isEqual(a, b)) return false;
 
-	let key;
+	a = objtools.deepCopy(a);
```

Comment:
I intentionally do an explicit copy rather than conditionally copying in `#subtractMask` if the result is a scalar.  Ask why if you want to know.